helm/cluster-azure/templates/_kcp.tpl

```diff
@@ -0,0 +1,192 @@
+            mountPath: /etc/kubernetes/azure.json
+            name: cloud-config
+            readOnly: true
+      scheduler:
+        extraArgs:
+          authorization-always-allow-paths: "/healthz,/readyz,/livez,/metrics"
+          bind-address: "0.0.0.0"
+      etcd:
+        local:
+          dataDir: /var/lib/etcddisk/etcd
+          extraArgs:
+            listen-metrics-urls: "http://0.0.0.0:2381"
+            quota-backend-bytes: "8589934592"
+      networking:
+        serviceSubnet: {{ .Values.network.serviceCIDR }}
+    diskSetup:
+      filesystems:
+        - device: /dev/disk/azure/scsi1/lun0
+          extraOpts:
+            - -E
+            - lazy_itable_init=1,lazy_journal_init=1
+          filesystem: ext4
+          label: etcd_disk
+        - device: ephemeral0.1
+          filesystem: ext4
+          label: ephemeral0
+          replaceFS: ntfs
+      partitions:
+        - device: /dev/disk/azure/scsi1/lun0
+          layout: true
+          overwrite: false
+          tableType: gpt
+    files:
+    {{- include "oidcFiles" . | nindent 4 }}
+    {{- include "sshFiles" . | nindent 4 }}
+    - contentFrom:
+        secret:
+          key: control-plane-azure.json
+          name: {{ include "resource.default.name" $ }}-control-plane-{{ include "hash" (dict "data" (include "controlplane-azuremachinetemplate-spec" $) "global" .) }}-azure-json
+      owner: root:root
+      path: /etc/kubernetes/azure.json
+      permissions: "0644"
+    - path: /etc/kubernetes/encryption/config.yaml
+      permissions: "0600"
+      contentFrom:
+        secret:
+          name: {{ include "resource.default.name" $ }}-encryption-provider-config
+          key: encryption
+    - path: /etc/kubernetes/policies/audit-policy.yaml
+      permissions: "0600"
+      encoding: base64
+      content: {{ $.Files.Get "files/etc/kubernetes/policies/audit-policy.yaml" | b64enc }}
+    initConfiguration:
+      nodeRegistration:
+        kubeletExtraArgs:
+          azure-container-registry-config: /etc/kubernetes/azure.json
+          cloud-config: /etc/kubernetes/azure.json
+          cloud-provider: external
+          feature-gates: CSIMigrationAzureDisk=true
+        name: '{{ `{{ ds.meta_data.local_hostname }}` }}'
+    joinConfiguration:
+      nodeRegistration:
+        kubeletExtraArgs:
+          azure-container-registry-config: /etc/kubernetes/azure.json
+          cloud-config: /etc/kubernetes/azure.json
+          cloud-provider: external
+          feature-gates: CSIMigrationAzureDisk=true
+        name: '{{ `{{ ds.meta_data.local_hostname }}` }}'
+    mounts:
+      - - LABEL=etcd_disk
+        - /var/lib/etcddisk
+    postKubeadmCommands: []
+    preKubeadmCommands: []
+  replicas: {{ .Values.controlPlane.replicas | default "3" }}
+  version: {{ .Values.kubernetesVersion }}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachineTemplate
+metadata:
+  labels:
+    cluster.x-k8s.io/role: control-plane
```

Comment:
should we add this label to the template `controlplane-azuremachinetemplate-spec` so that it also propagate to the `machine` object  ?

Reply:
good finding - yes we should do this (at least on AWS we already doing the same ([xref](https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/templates/_control_plane.tpl#L10C7-L11))